packages/xstate-solid/src/createImmutable.ts

```diff
@@ -1,25 +1,21 @@
-import { createStore, Store } from 'solid-js/store';
-import { deepClone } from './deepClone';
+import type { Store } from 'solid-js/store';
+import { createStore } from 'solid-js/store';
+import { deepClone, isWrappable } from './deepClone';
 import { batch } from 'solid-js';
 
-export function isWrappable(obj: any): obj is object {
-  let proto;
-  return (
-    obj != null &&
-    typeof obj === 'object' &&
-    (!(proto = Object.getPrototypeOf(obj)) ||
-      proto === Object.prototype ||
-      Array.isArray(obj))
-  );
-}
-
-const resolvePath = (path: any[], obj = {}) =>
-  path.reduce((prev, curr) => prev?.[curr], obj);
+const resolvePath = (path: any[], obj = {}) => {
+  let current = obj;
+  // tslint:disable-next-line:prefer-for-of
+  for (let i = 0; i < path.length; i++) {
+    current = current?.[path[i]];
+  }
+  return current;
+};
 
 const updateStore = <Path extends unknown[]>(
   nextStore: Store<any>,
   prevStore: Store<any>,
-  set: (...args: [...Path, any?, any?]) => void,
+  set: (...args: [...Path, unknown, unknown?]) => void,
```

Comment:
Ok, I see why you have used this type here now. It's a little bit of a workaround for TS quirks but I don't see how this could be improved. At least part of the problem is a bug~/incomplete behavior in TS. For instance, IMHO [this should be valid](https://www.typescriptlang.org/play?ts=4.8.2#code/GYVwdgxgLglg9mABFApgZygHgAqJQD1TABM1FwBrMOAdzAG0BdAPgAoAHALkWwBpE0KKN1YA6cQEMATgHM03euNF9yYKrTCMAlIgC8zRADc4MYjoDeAKEQ2BQsePb8AjIglkJYAJ78ATG49vLUsAXyA) but it isn't.